activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -381,14 +386,18 @@ def select_for_count
       end
 
       def build_count_subquery(relation, column_name, distinct)
-        column_alias = Arel.sql("count_column")
-        subquery_alias = Arel.sql("subquery_for_count")
+        relation.select_values = [
+          if column_name == :all
+            distinct ? table[Arel.star] : Arel.sql("1")
```

Comment:
Hi! I am the maintainer of the SQL Server adapter. I'm just catching up to this change and I found out it breaks all the adapter counts during this code path because this is not a valid syntax:

```sql
SELECT * FROM ( SELECT 1 ) count
```

It would be if `AS one` was used which is provided by `FinderMethods::ONE_AS_ONE`.